client/src/Movies/SavedList.js

```diff
@@ -1,22 +1,30 @@
-import React, { Component } from 'react';
+import React from 'react';
 import { Link, NavLink } from 'react-router-dom';
 
-export default class SavedList extends Component {
-  constructor(props) {
-    super(props);
-  }
+const SavedList = (props) => {
+  
+  function getUnique(arr, comp) {
+    const unique = arr
```

Comment:
wow, very creative with this solution, I would say to stay away from single letter params however as they'll be difficult to come back to later and understand what you were doing.  Also, I wonder if you could do this without having to chain so many methods? That is something to think about as though they're fast, if the data gets large enough, the input and algorithm time will start to stack up.